drivers/regex/mlx5/mlx5_regex.c

```diff
@@ -3,3 +3,15 @@
  */
 
 #include "mlx5_regex.h"
+
+int mlx5_regex_is_supported(struct ibv_context *ibv_ctx)
+{
+	struct mlx5_hca_attr caps;
+	int err;
+
+	err = mlx5_devx_cmd_query_hca_attr(ibv_ctx, &caps);
```

Comment:
The caps should be saved and reused, There is no point of reading the caps for each bit.
I assume that there will be other caps that we will need.